src/utils/utils.ts

```diff
@@ -220,6 +222,67 @@ function arrayZipWithPadding<T1, T2>(
   ]);
 }
 
+async function asyncIterableArray<T>(
+  iterable: AsyncIterable<T>,
+): Promise<Array<T>> {
+  const arr: Array<T> = [];
+  for await (const item of iterable) {
+    arr.push(item);
+  }
+  return arr;
+}
+
+function bufferSplit(
```

Comment:
This may not be necessary anymore if we are able to use `KeyPath` directly.

Primarily for composite keys where we wanted to have it sorted by a particular index, but still acquire the second part of the key.

But don't remove this function because it may be useful in the future.

Reply:
Look at `lexicographic level iteration order` test in js-db to see the rules of iteration that the DB does now, then you can see if it applies to what we want to use for composite keys.